core/src/app.rs

```diff
@@ -314,6 +318,9 @@ impl App {
 
         let project = self.project.as_ref().unwrap().clone();
         let run_id = self.run.as_ref().unwrap().run_id().to_string();
+        let mut secrets = self.secrets.as_ref().unwrap().clone();
+
+        secrets.retain(|k, _| k.to_uppercase() == *k);
```

Comment:
Let's instead validate that in the API route so that we error with a 400 when a secret is passed with a non upper case key?